pkg/kubectl/cmd/util/factory.go

```diff
@@ -165,6 +166,24 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+			if err != nil {
+				fmt.Printf("Failed to create client: %v", err)
+				return nil, nil
+			}
+			list, err := client.ThirdPartyResources(api.NamespaceDefault).List(api.ListOptions{})
```

Comment:
> extensions are included in the "v1" client...

Shouldn't we negotiate the client with `clients.ClientForVersion(&unversioned.GroupVersion{Group: "extensions"})` then?  Seems like we'd want to negotiate the version of the extensions group, not the legacy kube group.